apps/veyfi/contexts/useVotingEscrow.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {createContext, memo, useCallback, useContext, useMemo} from 'react';
+import {Contract} from 'ethcall';
+import {FixedNumber} from 'ethers';
+import useSWR from 'swr';
+import VEYFI_ABI from '@veYFI/utils/abi/veYFI.abi';
+import VEYFI_POSITION_HELPER_ABI from '@veYFI/utils/abi/veYFIPositionHelper.abi';
+import {VEYFI_ADDRESS, VEYFI_POSITION_HELPER_ADDRESS, YFI_ADDRESS} from '@veYFI/utils/constants';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import ERC20_ABI from '@yearn-finance/web-lib/utils/abi/erc20.abi';
+import {allowanceKey} from '@yearn-finance/web-lib/utils/address';
+import {toMilliseconds} from '@yearn-finance/web-lib/utils/time';
+import {getProvider, newEthCallProvider} from '@yearn-finance/web-lib/utils/web3/providers';
+
+import type {BigNumber} from 'ethers';
+import type {ReactElement} from 'react';
+import type {TAddress} from '@yearn-finance/web-lib/utils/address';
+import type {TMilliseconds} from '@yearn-finance/web-lib/utils/time';
+import type {TDict} from '@yearn-finance/web-lib/utils/types';
+
+export type TVotingEscrow = {
+	address: TAddress,
+	token: TAddress,
+	name: string,
+	symbol: string,
+	decimals: number,
+	supply: BigNumber,
+	rewardPool: TAddress,
+}
+
+export type TPosition = {
+	balance: BigNumber,
+	underlyingBalance: BigNumber,
+}
+
+export type TVotingEscrowPosition = {
+	deposit?: TPosition,
+	// yield?: TPosition,
+	unlockTime?: TMilliseconds,
+	penalty?: BigNumber,
+	penaltyRatio?: number,
+	withdrawable?: BigNumber,
+}
+
+export type	TVotingEscrowContext = {
+	votingEscrow: TVotingEscrow | undefined,
+	positions: TVotingEscrowPosition | undefined,
+	allowances: TDict<BigNumber>,
+	isLoading: boolean,
+	refresh: () => void,
+}
+const defaultProps: TVotingEscrowContext = {
+	votingEscrow: undefined,
+	positions: undefined,
+	allowances: {},
+	isLoading: true,
+	refresh: (): unknown => ({})
```

Comment:
I would probably remove the lint rule "Unexpected empty method", for these cases, so we don't need to cast the return to `unknown` in order to have defaults, /cc @Majorfi 
```typescript
const defaultProps: TVotingEscrowContext = {
	votingEscrow: undefined,
	positions: undefined,
	allowances: {},
	isLoading: true,
	// eslint-disable-next-line @typescript-eslint/no-empty-function
	refresh: (): void => {}
};
```

Reply:
Actually discussing with @Majorfi, we can do
```typescript
refresh: (): void => undefined
```
or even
```typescript
refresh: (): void => void(0)
```